File: cpp_codes/simpson.cc
#include <iostream>
#include <fstream>
#include <cmath>

using namespace std;

double func (double x) {
  double func = x;
  return func;
}

int main () {

  double a, b, h, I = 0;
  int n;

  cout << "Please enter integration limits: " << endl;
  cin >> a >> b >> n;

  h = (b-a)/n;
  I = I + func(a) + func(b);

  for (int i = 1; i < n; i++) {
    int weight = (n%2 ? 4 : 2);
    I = I + weight*func(a + i*h);
  }

  I = h*I/3.;

  cout << "The integration of the above function under limits " << a <<  " and " << b << " is: " << I << endl;
}

File: fortran_codes/simpson.f90
program simpson
  implicit none

  real :: a, b, h, I = 0
  integer :: n, ii, weight

  read(*,*) a, b, n

  h = (b-a)/n;
  I = I + func(a) + func(b);

  do ii = 1,n-1
    weight = merge(4, 2, mod(ii,2) == 1)
    I = I + weight*func(a + ii*h)
  end do

  I = h*I/3.
  print *, "The integration of the above function under limits", a, "and", b, "is:", I

contains

real function func (x)
  implicit none
  real :: x
  func = x**0.5
end function func

end program simpson
